### PR TITLE
Defaults only on specific fields

### PR DESCRIPTION
**Pull request by SneakyBerry**
Default for everything in schema is not included in spec of Avro
I have fixed it using "serde default" only for specific fields

### CODE REVIEW

**Comment by lerouxrgd** [comment]
Is there a situation where having default is bad ?

**Comment by SneakyBerry** [comment]
In my case yes.

For example I have schema

```json
{
  "type": "record",
  "name": "User",
  "namespace": "com.test",
  "fields": [ 
    {
      "name": "first_name",
      "type": "string"
    },
    {
        "name": "middle_name",
        "type": "string",
        "default": ""
    },
    {
        "name": "last_name",
        "type": "string"
    }
 ]
}
```
Due to this schema
this is correct:
```json
{ "first_name": "John", "last_name": "Doe" }
```

but this is not:
```json
{ "first_name": "John" }
```
And I want to get an error from Serde

File: src/templates.rs [hunk 1]
```diff
@@ -37,7 +37,6 @@ pub const RECORD_TEMPLATE: &str = r#"
 {%- endfor %}
 {%- endif %}
 #[derive(Debug, PartialEq, Clone, serde::Deserialize, serde::Serialize{%- if derive_builders %}, derive_builder::Builder {%- endif %})]
-#[serde(default)]
```
Comment (lerouxrgd) [inline src/templates.rs:40]: Could you check if `length(fields) == lenght(defaults)` and if it's true then keep `#[serde(default)]`.
Also keep the manual `impl Default`, something like:
```
impl Default for {{ name }} {
    fn default() -> {{ name }} {
        {{ name }} {
            {%- for f in fields %}
            {{ f }}: default_{{ name | lower }}_{{ f | lower | trim_start_matches(pat="r#") }}(),
            {%- endfor %}
        }
    }
}
```

File: src/templates.rs [hunk 2]
```diff
@@ -62,15 +64,12 @@ deser!(nullable_{{ name|lower }}_{{ f }}, {{ type }}, {{ defaults[f] }});
...
-        {{ name }} {
-            {%- for f in fields %}
-            {{ f }}: {{ defaults[f] }},
-            {%- endfor %}
-        }
-    }
-}
+{%- for f in fields %}
+{%- if defaults is containing(f) %}
+
```
[the remark this answers is not in the document]
Comment (SneakyBerry) [reply src/templates.rs:69]: It is for separation functions between each other

[the remark this answers is not in the document]
Comment (lerouxrgd) [reply src/templates.rs:69]: I think it's achievable with [whitespace control](https://tera.netlify.app/docs/#whitespace-control). If not, let's keep it then.

[the remark this answers is not in the document]
Comment (SneakyBerry) [reply src/templates.rs:69]: No, it can only trim whitespaces.

[the remark this answers is not in the document]
Comment (lerouxrgd) [reply src/templates.rs:69]: Maybe let's put an empty comment here then:

```
{%- if defaults is containing(f) %}
{# #}
fn default_{{ name | lower }}_{{ f | lower | trim_start_matches(pat="r#") }}() -> {{ types[f] }} { {{ defaults[f] }} }
{%- endif %}
```

File: src/gen.rs [hunk 3]
```diff
@@ -844,18 +761,9 @@ pub enum UnionStringAvroShortUuid {
...
 #[derive(Debug, PartialEq, Clone, serde::Deserialize, serde::Serialize)]
-#[serde(default)]
 pub struct AvroFileId {
     pub id: UnionStringAvroShortUuid,
 }
-
-impl Default for AvroFileId {
-    fn default() -> AvroFileId {
-        AvroFileId {
-            id: UnionStringAvroShortUuid::String(String::default()),
```
[the remark this answers is not in the document]
Comment (SneakyBerry) [reply src/gen.rs:855]: Done

File: src/templates.rs [hunk 4]
```diff
@@ -772,62 +780,57 @@ impl Templater {
...
                 .as_slice()
                 .join(", ");
-            format!("vec![{}]", vals)
+            Ok(format!("vec![{}]", vals))
         } else {
-            "vec![]".to_string()
-        };
-        Ok(default_str)
+            // "vec![]".to_string()
+            panic!()
```
[the remark this answers is not in the document]
Comment (SneakyBerry) [reply src/templates.rs]: Fixed

File: src/templates.rs [hunk 5]
```diff
@@ -772,62 +780,57 @@ impl Templater {
...
                     vals
-                )
+                ))
             }
         } else {
-            "::std::collections::HashMap::new()".to_string()
-        };
-        Ok(default_str)
+            // "::std::collections::HashMap::new()".to_string()
+            panic!()
```
[the remark this answers is not in the document]
Comment (SneakyBerry) [reply src/templates.rs]: Fixed

File: src/templates.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (SneakyBerry) [reply src/templates.rs]: Done

**Pull request merged**